gumroad/src/gumroad-block/block.js

```diff
@@ -0,0 +1,256 @@
+/**
+ * BLOCK: gumroad-block
+ *
+ * Registering a basic block with Gutenberg.
+ * Simple block, renders and saves the same content without any interactivity.
+ */
+
+//  Import CSS.
+import './style.scss';
+import './editor.scss';
+
+const { __ } = wp.i18n; // Import __() from wp.i18n
+const { Component } = wp.element;
+const { registerBlockType } = wp.blocks; // Import registerBlockType() from wp.blocks
+const {
+  InspectorControls,
+  RichText
+} = wp.editor;
+
+const {
+  SelectControl,
+  PanelBody
+} = wp.components
+
+const gumURL = 'https://gum.co/';
+const defaultID = 'DviQY';
+
+class GumControls extends Component {
+  constructor( props ) {
+    super( ...arguments );
+  }
+
+  compose_url( value ) {
+    if (value == '') {
+      return gumURL + defaultID;
+    }
+    return gumURL + value;
+  }
+
+  render() {
+    const typeOptions = [
+      { value: 'none', label: __( 'Link' )},
+      { value: 'overlay', label: __( 'Overlay' )},
+      { value: 'embed', label: __( 'Embed' )}
+    ];
+
+    const buttonOptions = [
+      { value: '', label: __( 'Link' )},
+      { value: 'gumroad-button', label: __( 'Button' )}
+    ];
+
+    const wantedOptions = [
+      { value: '', label: __( 'unwanted' )},
+      { value: 'true', label: __( 'Wanted' )}
+    ];
+
+    const { setAttributes, attributes: { id, type, button, classes, wanted, url } } = this.props;
+
+    return(
+      <InspectorControls key="GumControls">
+        <PanelBody>
+          <label>Product ID:</label>
+          <RichText
+            tagName="div"
+            placeholder={ __( defaultID ) }
+            value={ id }
+            onChange={ ( value ) => {
+                setAttributes( { id: value } );
+                setAttributes( { url: this.compose_url( value ) } );
+              }
+            }
+          /><br />
+          <SelectControl
+            label={ __( 'Behavior Type' ) }
+            description={ __( 'Set the type of Gumroad link behavior: Link, Overlay, or Embed.' ) }
+            options={ typeOptions }
+            value={ type }
+            onChange={ ( value ) => setAttributes( { type: value } ) }
+            />
+          <SelectControl
+            label={ __( 'Button?' ) }
+            description={ __( 'Make the link a button.' ) }
+            options={ buttonOptions }
+            value={ button }
+            onChange={ ( value ) => setAttributes( { button: value } ) }
+            />
+          <SelectControl
```

Comment:
should be a BOOL (checkbox) and the label can be "Auto-trigger the payment form"

Reply:
This is done. Looks much better too.